### PR TITLE
Allow reconnect active connection to multiple providers

### PR DESCRIPTION
**Pull request by soffokl**
This is an essential part of the smart-reconnect. 
Closes https://github.com/mysteriumnetwork/node/issues/3247 and part of the https://github.com/mysteriumnetwork/node/issues/3870 

The new flag was introduced to explicitly enable reconnect: `--auto-reconnect=true` (disabled by default).

Currently it should reconnect if both flags `--auto-reconnect=true` and `--keep-connected-on-fail=true` set and multiple providers passed for connection create:
`myst connection up 0xe948dae2ce1faf719ba1091d8c6664a46bab073d,0x070a908615c53ff0aa69c4e802142791c7d9ef09`

### CODE REVIEW

File: core/connection/manager.go [hunk 1]
```diff
@@ -834,7 +853,14 @@ func (m *connectionManager) setupTrafficBlock(disableKillSwitch bool) error {
 }
 
 func (m *connectionManager) reconnectOnHold(state connectionstate.AppEventConnectionState) {
-	if state.State == connectionstate.StateOnHold {
+	if state.State == connectionstate.StateOnHold && config.GetBool(config.FlagAutoReconnect) {
```
Comment (vkuznecovas) [inline core/connection/manager.go:856]: can we unnest this and just return if state != on hold or not auto reconnect?

File: router/router.go [hunk 2]
```diff
@@ -18,44 +18,210 @@
...
+
+func (m *manager) ExcludeIP(ip net.IP) error {
+	m.ensureStarted()
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	new := true
+
+	for i, rule := range m.rules {
+		if rule.ip.Equal(ip) {
```
Comment (vkuznecovas) [inline router/router.go:80]:
```suggestion
		if !rule.ip.Equal(ip) { 
		  continue
		}
		...
```

File: router/router.go [hunk 3]
```diff
@@ -18,44 +18,210 @@
...
+
+	return nil
+}
+
+func (m *manager) RemoveExcludedIP(ip net.IP) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	for i, rule := range m.rules {
+		if rule.ip.Equal(ip) {
```
Comment (vkuznecovas) [inline router/router.go:110]:
```suggestion
		if !rule.ip.Equal(ip) { 
		  continue
		}
		...
```

File: router/router.go [hunk 4]
```diff
@@ -18,44 +18,210 @@
...
 		}
+	}
+}
 
+func (m *manager) Stop() {
+	if err := m.Clean(); err != nil {
+		log.Error().Err(err).Msg("Failed to clean routing rules")
 	}
+
+	close(m.stop)
```
Comment (vkuznecovas) [inline router/router.go:152]: wrap this in a once, since we're always stopping twice :D

**Review by vkuznecovas — APPROVED** [review]

**Comment by Snawoot** [comment]
Doesn't work for some reason:

```
» connect 0xa3caf88de5ac00d00876039e7f563416b53d0a78 0x18730b1a9864b959a6edd1b7e715ddef5f835b00,0xcdfdba200b2455c503406deae7809d90b886d4c9 wireguard
[CONNECTING] from: 0xa3caf88de5ac00d00876039e7f563416b53d0a78 to: 0x18730b1a9864b959a6edd1b7e715ddef5f835b00,0xcdfdba200b2455c503406deae7809d90b886d4c9
[ERROR] Server response invalid: 500 Internal Server Error (http://127.0.0.1:4050/connection). Possible error: failed to lookup proposal: ErrorCollection: proposal does not exist: {ServiceType:wireguard ProviderID:0x18730b1a9864b959a6edd1b7e715ddef5f835b00,0xcdfdba200b2455c503406deae7809d90b886d4c9 ID:0}
```

**Comment by soffokl** [comment]
@Snawoot you read it wrong. It should be:
```
myst connection up 0x18730b1a9864b959a6edd1b7e715ddef5f835b00,0xcdfdba200b2455c503406deae7809d90b886d4c9
```

It's not added to the interactive CLI yet.

**Review by Snawoot — APPROVED** [review]

**Pull request merged**